lib/request.js

```diff
@@ -168,11 +183,19 @@ Object.defineProperties(Request.prototype, {
   routeOptions: {
     get () {
       const context = this[kRouteContext]
-      const serverLimit = context.server.initialConfig.bodyLimit
       const routeLimit = context._parserOptions.limit
-      return {
-        bodyLimit: ((routeLimit || serverLimit))
+      const serverLimit = context.server.initialConfig.bodyLimit
+      const options = {
+        method: context.config.method,
+        url: this.raw.url,
+        bodyLimit: (routeLimit || serverLimit),
+        attachValidation: context.attachValidation,
+        logLevel: context.logLevel
       }
+      return readOnly(options)
```

Comment:
Why define a new function when `Object.freeze` exists?

Reply:
My original suggestion to use `Proxy` is mainly because I want to reduce the maintenance on future property added.
`ReadOnly(context) { return new Proxy() }` and remove the unnecessary property.

Yes, if it is a new object here. `Object.freeze` do better job.

Reply:
done 👍